### ISSUE TITLE
ComfyUI has removed  batch_area_memory()

### ISSUE THREAD

**Issue by Lerc** [SeargeDP/SeargeSDXL#103]
https://github.com/comfyanonymous/ComfyUI/commit/dd4ba68b6e93a562d9499eff34e50dbbbc8714e7

batch_area_memory(area) has gone and seemingly replaced with  memory_required(self, input_shape)

**Comment by Michedev** [comment]
Even if I copy paste back the function I got another error:

```
  File "/home/devita/stable-diffusion/ComfyUI/execution.py", line 153, in recursive_execute
    output_data, output_ui = get_output_data(obj, input_data_all)
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/execution.py", line 83, in get_output_data
    return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/execution.py", line 76, in map_node_over_list
    results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/custom_nodes/SeargeSDXL/modules/magic_box.py", line 269, in process
    (data, stage_result) = self.run_stage(stage, data, stage_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/custom_nodes/SeargeSDXL/modules/magic_box.py", line 248, in run_stage
    (data, stage_result) = stage_processor.process(data, stage_input)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/custom_nodes/SeargeSDXL/modules/stage_sampling.py", line 126, in process
    latent = sampler(base_model, base_positive, base_negative, latent, seed, steps, cfg,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/custom_nodes/SeargeSDXL/modules/node_wrapper.py", line 94, in sdxl_sampler
    result = sdxl_ksampler(base_model, None, noise_seed, base_steps, 0, cfg, sampler_name,
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/custom_nodes/SeargeSDXL/modules/custom_sdxl_ksampler.py", line 356, in sdxl_ksampler
    samples = sdxl_sample(base_model, refiner_model, noise, base_steps, refiner_steps, cfg, sampler_name, scheduler,
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/devita/stable-diffusion/ComfyUI/custom_nodes/SeargeSDXL/modules/custom_sdxl_ksampler.py", line 184, in sdxl_sample
    pos_base_copy = comfy.sample.broadcast_cond(base_positive, noise.shape[0], device)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: module 'comfy.sample' has no attribute 'broadcast_cond'
```

**Comment by iGerman00** [comment]
Rewriting it to the commit also doesn't work:
```python
    memory_required = base_model.memory_required(noise.shape[0] * noise.shape[2] * noise.shape[3]) + inference_memory
```

@SeargeDP looks like you need another update to make it compatible with the latest ComfyUI.

**Comment by Lerc** [comment]
@iGerman00 looks like memory_required takes a shape  so perhaps try

`memory_required = base_model.memory_required(noise.shape) + inference_memory`

**Comment by kevincao8828** [comment]
Traceback (most recent call last):
  File "/home/ps/ComfyUI/nodes.py", line 1735, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/ps/ComfyUI/custom_nodes/SeargeSDXL/__init__.py", line 33, in <module>
    from .modules._legacy import LEGACY_CLASS_MAPPINGS, LEGACY_DISPLAY_NAME_MAPPINGS
  File "/home/ps/ComfyUI/custom_nodes/SeargeSDXL/modules/__init__.py", line 29, in <module>
    from .ui import Defs
  File "/home/ps/ComfyUI/custom_nodes/SeargeSDXL/modules/ui.py", line 37, in <module>
    from .custom_sdxl_ksampler import CfgMethods
  File "/home/ps/ComfyUI/custom_nodes/SeargeSDXL/modules/custom_sdxl_ksampler.py", line 38, in <module>
    from comfy.model_management import get_torch_device, batch_area_memory, load_models_gpu
ImportError: cannot import name 'batch_area_memory' from 'comfy.model_management' (/home/ps/ComfyUI/comfy/model_management.py)

Cannot import /home/ps/ComfyUI/custom_nodes/SeargeSDXL module for custom nodes: cannot import name 'batch_area_memory' from 'comfy.model_management' (/home/ps/ComfyUI/comfy/model_management.py)

**Comment by toon-geek** [comment]
@kevincao8828 
getting the same import error

**Comment by SeargeDP** [comment]
I have a fix for that on the dev/test branch. If you are using the install script you can use it to switch to that branch and test the fix. Once I release the update with this fix I recommend to switch back to the main branch.

**Comment by SeargeDP** [comment]
The fix is now on the main branch and has been released in v4.3.2
